x-pack/plugins/security_solution/public/common/components/markdown/index.test.tsx

```diff
@@ -1,176 +0,0 @@
-
-    test('it applies EUI table styling to tables', () => {
-      const wrapper = mount(<Markdown raw={rawTable} />);
-
-      expect(wrapper.find('table').first().childAt(0).hasClass('euiTable')).toBe(true);
-    });
-
-    headerColumns.forEach((headerText) => {
-      test(`it renders the "${headerText}" table header`, () => {
-        const wrapper = mount(<Markdown raw={rawTable} />);
-
-        expect(wrapper.find('[data-test-subj="markdown-table-header"]').first().text()).toContain(
-          headerText
-        );
-      });
-    });
-
-    test('it applies EUI table styling to table rows', () => {
-      const wrapper = mount(<Markdown raw={rawTable} />);
-
-      expect(
-        wrapper
-          .find('[data-test-subj="markdown-table-row"]')
-          .first()
-          .childAt(0)
-          .hasClass('euiTableRow')
-      ).toBe(true);
-    });
-
-    test('it applies EUI table styling to table cells', () => {
-      const wrapper = mount(<Markdown raw={rawTable} />);
-
-      expect(
-        wrapper
-          .find('[data-test-subj="markdown-table-cell"]')
-          .first()
-          .childAt(0)
-          .hasClass('euiTableRowCell')
-      ).toBe(true);
-    });
-
-    test('it renders the expected table content', () => {
-      const wrapper = shallow(<Markdown raw={rawTable} />);
-
-      expect(wrapper).toMatchSnapshot();
-    });
-  });
-
-  describe('markdown links', () => {
-    const markdownWithLink = 'A link to an external site [External Site](https://google.com)';
-
-    test('it renders the expected link text', () => {
-      const wrapper = mount(<Markdown raw={markdownWithLink} />);
-
-      expect(wrapper.find('[data-test-subj="markdown-link"]').first().text()).toEqual(
-        'External Site'
-      );
-    });
-
-    test('it renders the expected href', () => {
-      const wrapper = mount(<Markdown raw={markdownWithLink} />);
-
-      expect(wrapper.find('[data-test-subj="markdown-link"]').first().getDOMNode()).toHaveProperty(
-        'href',
-        'https://google.com/'
-      );
-    });
-
-    test('it does NOT render the href if links are disabled', () => {
-      const wrapper = mount(<Markdown disableLinks={true} raw={markdownWithLink} />);
-
-      expect(
-        wrapper.find('[data-test-subj="markdown-link"]').first().getDOMNode()
-      ).not.toHaveProperty('href');
-    });
-
-    test('it opens links in a new tab via target="_blank"', () => {
-      const wrapper = mount(<Markdown raw={markdownWithLink} />);
-
-      expect(wrapper.find('[data-test-subj="markdown-link"]').first().getDOMNode()).toHaveProperty(
-        'target',
-        '_blank'
-      );
-    });
-
-    test('it sets the link `rel` attribute to `noopener` to prevent the new page from accessing `window.opener`, `nofollow` to note the link is not endorsed by us, and noreferrer to prevent the browser from sending the current address', () => {
```

Comment:
The intent of the behavior being asserted in this test is to prevent links to the new pages from accessing `window.opener`, which is a vector for [Tabnabbing](https://owasp.org/www-community/attacks/Reverse_Tabnabbing).

Would you be willing to restore this behavior and add an equivalent test?

Reply:
[The EUI automatically adds](https://github.com/elastic/eui/blob/master/src/services/security/get_secure_rel_for_target.ts#L26) the noopener and nofollow to [all links](https://github.com/elastic/eui/search?q=getSecureRelForTarget) used by EUI including those in the [markdown](https://github.com/elastic/eui/blob/master/src/components/markdown_editor/plugins/markdown_default_plugins.tsx#L77).

Reply:
> [The EUI automatically adds](https://github.com/elastic/eui/blob/master/src/services/security/get_secure_rel_for_target.ts#L26) the noopener and nofollow to [all links](https://github.com/elastic/eui/search?q=getSecureRelForTarget) used by EUI including those in the [markdown](https://github.com/elastic/eui/blob/master/src/components/markdown_editor/plugins/markdown_default_plugins.tsx#L77). In my opinion there is no need to test their implementation.

I"m not seeing the behavior described above when desk testing. Per the screenshot below:

🔴 the link is missing `noopener `
🔴 the link is missing `nofollow`
🟢 the link has `noreferrer`

<img width="927" alt="links" src="https://user-images.githubusercontent.com/4459398/96149758-ea2c0900-0ec6-11eb-90e2-f55f520a94a1.png">

Reply:
The link is missing the `noopener` and `nofollow` because the link in the screenshot does not have the attribute `target=_blank`. Only links that open to a new tab are vulnerable to Tabnabbing and need these extra properties.

Reply:
We're on the same page now. The change in behavior, where links no longer have `target="_blank"`, was an unexpected side effect of this PR.